### DISCUSSION TITLE
Configuring MaxPendingReceives in CoreWCF WebHttp endpoint

### DISCUSSION THREAD

**Discussion by LijoAlex87** [CoreWCF/CoreWCF#1094] [General]
I am migrating a WCF service to CoreWCF and have the following behavior

```
    <endpointBehaviors>
        <behavior name="MultiReceive">
          <dispatcherSynchronization maxPendingReceives="30" />
        </behavior>
      </endpointBehaviors>
```

Is there a way I can configure this in CoreWCF ? WCF used to have https://learn.microsoft.com/en-us/dotnet/api/system.servicemodel.description.dispatchersynchronizationbehavior?view=netframework-4.7 . Is there something similar I can use in CoreWCF

Also, I need to add this to an endpoint supporting WebHttpBinding. Have been following https://corewcf.github.io/blog/2022/04/13/webhttp . Is there a way I can do it in WebHttpBehavior itself while configuring ServiceEndpoint via the delegate.
Something like 

`serviceBuilder.AddServiceWebEndpoint<MyService, IMyService>(new WebHttpBinding(), "s", (behavior) => {behavior.MaxPendingReceieves = 30 ; } );`

Let me know how I can acheive this

**Comment by LijoAlex87** [comment]
Inorder to do this I created a new Behaviour which sets MaxPendingReceives as below

```
    public class MultiReceiveBehavior : Attribute, IEndpointBehavior
    {
        #region IServiceBehavior Members

        /// <inheritdoc/>
        public void AddBindingParameters(ServiceEndpoint endpoint, BindingParameterCollection bindingParameters)
        {
        }

        /// <inheritdoc/>
        public void ApplyClientBehavior(ServiceEndpoint endpoint, ClientRuntime clientRuntime)
        {
        }

        /// <inheritdoc/>
        public void ApplyDispatchBehavior(ServiceEndpoint endpoint, EndpointDispatcher endpointDispatcher)
        {
            endpointDispatcher.ChannelDispatcher.MaxPendingReceives = 30;
        }

        /// <inheritdoc/>
        public void Validate(ServiceEndpoint endpoint)
        {
        }

        #endregion
    }
```

But I am not able to figure out how can I add Endpint Behavior to an endpoint supporting WebHttpBinding. I know I can do the below for a regular SOAP endpoint

`builder.AddServiceEndpoint<EWSService>(typeof(MyContract), new BasicHttpBinding(), "http://localhost:5000/MyService", (endpoint) => { endpoint.EndpointBehaviors.Add(new MultiReceiveBehavior()); });`

But I cannot do this for endpoint supporting WebHttpBinding

`serviceBuilder.AddServiceWebEndpoint<MyService, MyContract>(new WebHttpBinding(), "s");`  

Since AddServiceWebEndpoint definition is of the form below and dont expose endpoint in delegate unlike AddServiceEndpoint

 public static IServiceBuilder AddServiceWebEndpoint<TService, TContract>(
            this IServiceBuilder builder,
            WebHttpBinding binding,
            string address,
            Action<WebHttpBehavior> configureWebBehavior = null) => AddServiceWebEndpoint<TService>(
                builder,
                typeof(TContract),
                binding,
                address,
                configureWebBehavior);

Considering this what are my options to support maxPendingReceives ? Is it supported for CoreWCF WebHttp endpoint

**Reply by g7ed6e** [reply]
@LijoAlex87 I just checked the code base and it looks like despite MaxPendingReceives property is public, setting the property value is inhibited and the value is hardcoded to 1. Please have a look to [ChannelHandler](https://github.com/CoreWCF/CoreWCF/blob/8aad8508d82a65bc6ca79178943321fa9bf70f24/src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/ChannelHandler.cs#L62) and [ChannelDispatcher](https://github.com/CoreWCF/CoreWCF/blob/8aad8508d82a65bc6ca79178943321fa9bf70f24/src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/ChannelDispatcher.cs#L75).

**Reply by LijoAlex87** [reply]
Thanks a lot for looking into this
